### PR TITLE
[MM-43075] Channel Info RHS: shortcut

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
Add a shortcut to toggle channel info RHS: CTRL(or cmd)+Alt+I

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43075

#### Related Pull Requests
N/A

#### Screenshots
2022-04-12-160456_380x165_scrot

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by pvev — APPROVED** [review]
LGTM! nice work @JulienTant :shipit:

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Looks good @JulienTant. The only request I have is to make the shortcut act as a toggle so that it also closes it if you hit the shortcut again while it's open

**Comment by stevemudie** [comment]
Looks good @JulienTant!

+1 for the toggle

I'll test and pass it once that's in place.

**Review by matthewbirtch — APPROVED** [review]
LGTM!

**Comment by stevemudie** [comment]
@JulienTant I took a second pass with this yesterday and found some issues that I'd missed the first time around when testing only on my Mac.

The biggest one is that **Ctrl+Shift+I** is the default shortcut for opening the Developer Tools on Chrome on Windows.
Chrome dev tools
As such using those keys would toggle that open and closed some of the time while opening and closing the info at other times.

Also, while testing on a Windows 10 PC the **Ctrl+.** only worked if you press the dot on the number pad as it ignored the one on the main keyboard. On Windows 7 (using a different PC) it worked as expected but then would only work to expand (**Ctrl+shift+.**) when using the one on the main keyboard (and not the number pad one).

Meanwhile on both macOS and Windows (7 & 10) when you use **Ctrl/CMD+.** it tosses up two identical 400 errors each time you use it which might be an issue?
Error 400

File: e2e/cypress/tests/integration/keyboard_shortcuts/ctrl_cmd_I_toggles_channel_info_rhs_spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+describe('Keyboard Shortcuts', () => {
+    let url;
+
+    before(() => {
+        cy.apiInitSetup({loginAfter: true}).then(({channelUrl}) => {
+            url = channelUrl;
+        });
+    });
+
+    it('CTRL/CMD+SHIFT+I - Toggle Channel Info RHS', () => {
```
Comment (stevemudie) [inline e2e/cypress/tests/integration/keyboard_shortcuts/ctrl_cmd_I_toggles_channel_info_rhs_spec.js:22]:
```suggestion
    it('MM-T4775 CTRL/CMD+SHIFT+I - Toggle Channel Info RHS', () => {
```

Comment (stevemudie) [reply e2e/cypress/tests/integration/keyboard_shortcuts/ctrl_cmd_I_toggles_channel_info_rhs_spec.js:22]: I've added the test case for this at https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4775

Comment (JulienTant) [reply e2e/cypress/tests/integration/keyboard_shortcuts/ctrl_cmd_I_toggles_channel_info_rhs_spec.js]: Cool thanks!

**Comment by JulienTant** [comment]
> The biggest one is that **Ctrl+Shift+I** is the default shortcut for opening the Developer Tools on Chrome on Windows. As such using those keys would toggle that open and closed some of the time while opening and closing the info at other times.

I can't seem to be able to reproduce the issue. There are conditions for it to show tho: you have to be in a channel (or gm/dm) for the shortcut to be registered, it won't work on the login page, or in boards, .... If you have a way to make it fail for sure feel free to tell me. Gentle ping to @matthewbirtch for the shortcut: maybe the CTRL+SHIFT+I shortcut will end up being annoying for devs who want to display the dev console.

> Meanwhile on both macOS and Windows (7 & 10) when you use **Ctrl/CMD+.** it tosses up two identical 400 errors each time you use it which might be an issue?

It looks like I forgot to register the Channel Info RHS as the previously opened panel, I'll push a fix for this!

> Also, while testing on a Windows 10 PC the Ctrl+. only worked if you press the dot on the number pad as it ignored the one on the main keyboard. On Windows 7 (using a different PC) it worked as expected but then would only work to expand (Ctrl+shift+.) when using the one on the main keyboard (and not the number pad one).

Is it possible to make a ticket for the webapp/channels team?

**Comment by JulienTant** [comment]
> Meanwhile on both macOS and Windows (7 & 10) when you use Ctrl/CMD+. it tosses up two identical 400 errors each time you use it which might be an issue?

Fixed after the last push: after opening the channel info RHS, `CTRL+.` will now close and reopen the channel info rhs.

**Comment by stevemudie** [comment]
@JulienTant The 400 error is no longer there so that's all set. 👍🏻

I tested further using Windows 11 on my Mac (with Parallels) and it worked as expected (including when using an external keyboard with number keys). The only problem being that it wouldn't open the dev tools on Chrome which only matters to devs, QA and a few others. I couldn't get it to trigger the dev tools except when I was on Playbooks or Boards which is expected.

**Comment by JulienTant** [comment]
@stevemudie are we ok to merge this then?

**Comment by stevemudie** [comment]
> @stevemudie are we ok to merge this then?

@JulienTant I'm fine with it but the only reason I was holding off was to see what @matthewbirtch had to say about the Windows Chrome conflict. As a guess he's out today.

**Comment by matthewbirtch** [comment]
> > The biggest one is that **Ctrl+Shift+I** is the default shortcut for opening the Developer Tools on Chrome on Windows. As such using those keys would toggle that open and closed some of the time while opening and closing the info at other times.
> 
> I can't seem to be able to reproduce the issue. There are conditions for it to show tho: you have to be in a channel (or gm/dm) for the shortcut to be registered, it won't work on the login page, or in boards, .... If you have a way to make it fail for sure feel free to tell me. Gentle ping to @matthewbirtch for the shortcut: maybe the CTRL+SHIFT+I shortcut will end up being annoying for devs who want to display the dev console.

Ya, that's a pretty big conflict. I guess I missed that. We should definitely find an alternative for the windows shortcut. What about `Cltr` `Alt` `I`?

**Comment by JulienTant** [comment]
I changed  Shift to Alt 👍🏻

**Review by matthewbirtch — APPROVED** [review]

**Review by stevemudie — APPROVED** [review]
Looks good @JulienTant 

I wasn't able to get Ctrl+Alt+I to work on Windows11 so I asked around to see if someone could test it on an actual PC and it's working fine (I use Parallels on a Mac).

**Pull request merged**